data-runtime/src/main/java/io/micronaut/data/runtime/mapper/DTOMapper.java

```diff
@@ -109,20 +109,27 @@ public Object read(@NonNull S object, @NonNull String name) throws ConversionErr
         }
         pp = dtoEntity.getPropertyByName(name);
         if (pp == null) {
-            throw new DataAccessException("DTO projection doesn't defines a property [" + name + "] on DTO entity: " + dtoEntity.getName());
+            throw new DataAccessException("DTO projection doesn't define a property [" + name + "] on DTO entity: " + dtoEntity.getName());
         }
         return read(object, pp);
     }
 
     @Nullable
     @Override
     public Object read(@NonNull S object, @NonNull Argument<?> argument) {
-        RuntimePersistentProperty<T> pp = persistentEntity.getPropertyByName(argument.getName());
+        String name = argument.getName();
+        RuntimePersistentProperty<T> pp = persistentEntity.getPropertyByName(name);
         if (pp == null) {
+            if (persistentEntity != dtoEntity) {
```

Comment:
Try to find persistent property in DTO entity if prop not found in mapped entity.